### ISSUE TITLE
Missing WILDFLY_HOME in /etc/wildfly/wildfly.conf

### ISSUE THREAD

**Issue by rgarrigue** [biemond/biemond-wildfly#161]
Hi

I'm calling wildfly class like this, with profile::wildfly::nom = 'wf10' for unitary test purpose

```
  class { '::wildfly':
    user           => $::profiles::wildfly::nom,
    group          => $::profiles::wildfly::nom,
    dirname        => "/opt/${::profiles::wildfly::nom}",
    service_name   => $::profiles::wildfly::nom,
    version        => $::profiles::wildfly::version,
    install_source => "http://download.jboss.org/wildfly/${::profiles::wildfly::version}.Final/wildfly-${::profiles::wildfly::version}.Final.tar.gz",
    java_home      => "/usr/lib/jvm/${::profiles::wildfly::java}",
    java_opts      => $::profiles::wildfly::java_opts,
  }
```

I got a puppet error deploying a resource

```
...
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Prepare/File[/opt/wf10]/ensure: created
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Install/Exec[Download wildfly from http://download.jboss.org/wildfly/10.1.0.Final/wildfly-10.1.0.Final.tar.gz]/returns: executed successfully
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Install/Exec[untar wildfly-10.1.0.Final.tar.gz]/returns: executed successfully
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Install/Exec[untar wildfly-10.1.0.Final.tar.gz]: Triggered 'refresh' from 1 events
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Setup/File[/opt/wf10/jboss.properties]/ensure: defined content as '{md5}9ad08c0c6918594eeef1a469bc09c120'
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Setup/Wildfly::Config::Mgmt_user[puppet]/Wildfly::Config::User[mgmt_puppet]/File_line[puppet:ManagementRealm]/ensure: created
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Setup/File[/opt/wf10/bin/standalone.conf]/content: content changed '{md5}17fc243b866d997913cea398137fc3aa' to '{md5}85a7c6bb8e4282d8b7a0763b45264ad8'
==> testcentos.poc: Notice: /File[/opt/wf10/bin/standalone.conf]/seluser: seluser changed 'unconfined_u' to 'system_u'
==> testcentos.poc: Notice: /File[/opt/wf10/bin/standalone.conf]/seltype: seltype changed 'usr_t' to 'bin_t'
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Service::Systemd/File[/opt/wf10/bin/launch.sh]/ensure: defined content as '{md5}e83fdd005ebd4384496b08a90d940510'
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Service/File[/etc/wildfly]/ensure: created
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Service/File[/etc/wildfly/wildfly.conf]/ensure: created
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Service::Systemd/File[/etc/systemd/system/wf10.service]/ensure: created
==> testcentos.poc: Notice: /Stage[main]/Wildfly::Service/Service[wf10]/ensure: ensure changed 'stopped' to 'running'
==> testcentos.poc: Error: /Stage[main]/Profiles::Wildfly/Wildfly::Resource[/subsystem=datasources/data-source=outileleveur]/Wildfly_resource[/subsystem=datasources/data-source=outileleveur]: Could not evaluate: Connection refused - connect(2)
```

Connected to the VM, I see the wf10 service not working

```
[root@testcentos ~]# systemctl status wf10
● wf10.service - The WildFly Application Server
   Loaded: loaded (/etc/systemd/system/wf10.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since jeu. 2016-12-08 11:00:11 CET; 4s ago
  Process: 29874 ExecStart=/opt/wf10/bin/launch.sh $WILDFLY_MODE $WILDFLY_CONFIG $WILDFLY_BIND (code=exited, status=127)
 Main PID: 29874 (code=exited, status=127)

déc. 08 11:00:11 testcentos.poc.adventiel.io systemd[1]: Started The WildFly Application Server.
déc. 08 11:00:11 testcentos.poc.adventiel.io systemd[1]: Starting The WildFly Application Server...
déc. 08 11:00:11 testcentos.poc.adventiel.io systemd[1]: wf10.service: main process exited, code=exited, status=127/n/a
déc. 08 11:00:11 testcentos.poc.adventiel.io systemd[1]: Unit wf10.service entered failed state.
déc. 08 11:00:11 testcentos.poc.adventiel.io systemd[1]: wf10.service failed.
```

I added this little stuff in launch.sh

```
echo "/opt/wf10/bin/launch.sh" > /tmp/debug.log
env | grep WILDFLY >> /tmp/debug.log
``` 

And I see there's not WILDFLY_HOME in the debug.log

```
[root@testcentos ~]# cat /tmp/debug.log 
/opt/wf10/bin/launch.sh
WILDFLY_CONFIG='standalone.xml -P /opt/wf10/jboss.properties'
WILDFLY_MODE=standalone
WILDFLY_BIND=0.0.0.0
``` 

Which is a problem as default is /opt/wildfly and mine is /opt/wf10. Adding it to /etc/wildfly/wildfly.conf solved this step

**Comment by jairojunior** [comment]
You're right. `launch.sh` should be a template. :+1: 

This script used to be copied from official distribution: https://github.com/wildfly/wildfly-core/blob/master/core-feature-pack/src/main/resources/content/docs/contrib/scripts/systemd/launch.sh

But I had to bring this to my module in order to fix it (it declares /bin/sh, but only works with /bin/bash). I'll fix it.

**Comment by jairojunior** [comment]
Fixed in v1.0.1. Thanks again.
